### PR TITLE
Voting

### PR DESCRIPTION
**Pull request by pjazdzewski1990**
MVP for voting 
It still lacks few things (persistent sessions, command to display status), but I would like to get this in front of people so they can give me few hints on how the interaction should look like

### CODE REVIEW

File: src/main/scala/io/scalac/slack/bots/voting/VotingBot.scala [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+      val response = repo.findSession(sessionId) match {
+        case Some(session) =>
+          repo.closeSession(sessionId, session)
+          
+          val votingResults = session.votes.groupBy(_.answer).map{ case (answerId, votes) =>
+            val answerText = session.topic.answers(answerId)
+            val voters = votes.foldLeft("")((acc, v) =>  acc + mention(v.voter))
+            (s"$answerText VOTES #${votes.length} by $voters", votes.length)
+          }.toList.sortBy(_._2).map(_._1)
```
Comment (marioosh) [inline src/main/scala/io/scalac/slack/bots/voting/VotingBot.scala]: może wyniki też prezentować? chociaż gdzieś tam w nawiasie?

File: src/main/scala/io/scalac/slack/bots/voting/VotingBot.scala [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    s"${user}: Session $sessionId has been closed"
+  }
+
+  def formatNoAnswerMessage(sessionId: Long, user: String, answerId: Int): String = {
+    s"${user}: Session $sessionId has no answer $answerId"
+  }
+
+  def formatNoSessionMessage(sessionId: Long, user: String): String = {
+    s"${user}: No session with this Id: $sessionId"
+  }
```
Comment (marioosh) [inline src/main/scala/io/scalac/slack/bots/voting/VotingBot.scala]: Skip `:String` as return type and bracers, make it oneliners with short functions. `${user}` replace with `$user`

File: src/main/scala/io/scalac/slack/bots/voting/VotingRepo.scala [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  import VoteResult._
+
+  var globalVotingId = 0L
+  val sessionStorage = mutable.Map[Long, Session]()
+
+  override def findSession(sessionId: Long): Option[Session] = {
+    sessionStorage.get(sessionId)
+  }
+
+  override def addVote(sessionId: Long, vote: Vote): VoteResult.Value = {
```
Comment (marioosh) [inline src/main/scala/io/scalac/slack/bots/voting/VotingRepo.scala]: wraping `{` is not necessary here

File: src/main/scala/io/scalac/slack/bots/voting/VotingRepo.scala [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    findSession(sessionId) match {
+      case Some(session) if !session.topic.isOpened =>
+        SessionClosed
+      case Some(session) if session.topic.answers.length > vote.answer =>
```
Comment (marioosh) [inline src/main/scala/io/scalac/slack/bots/voting/VotingRepo.scala]: `>` is in opposite direction here. Number of possible answers should be always higher than answered number.

Comment (pjazdzewski1990) [inline src/main/scala/io/scalac/slack/bots/voting/VotingRepo.scala]: No. `vote.answer` is the number user put on the UI, `session.topic.answers.length` is the amount of all possible answers. For instance if the input is 3 and there are 3 options to choose from (0, 1, 2), this condition should fail

File: src/main/scala/io/scalac/slack/bots/voting/VotingBot.scala (hunk 1 above)
Comment (pjazdzewski1990) [inline src/main/scala/io/scalac/slack/bots/voting/VotingBot.scala]: All the answers that were given at leas one vote will be presented (`session.votes`).
They will be presented from the one/ones having most votes.
Response will hold text, number of votes and who voted for each non-empty answer

File: src/main/scala/io/scalac/slack/bots/voting/VotingRepo.scala (hunk 4 above)
Comment (marioosh) [inline src/main/scala/io/scalac/slack/bots/voting/VotingRepo.scala]: yes, Like I said :) if there are only 3 possible answers (`session.topic.anwers.length` is 3) you can answer with (0,1,2)
Assume votes:
vote 0 : `3 > 0 => NoAnswer`
vote 1 : `3 > 1 => NoAnswer`
vote 2 : `3 > 2 => NoAnswer`

Lets try with incorrect vote number:
vote 3 : `3 > 3 => ` false so, vote 3 is passed to the next match statement.

So, `NoAnswer` should be produced only for 3 and more - `vote.answer` is equal or greater than number of possible answers.

Comment (pjazdzewski1990) [inline src/main/scala/io/scalac/slack/bots/voting/VotingRepo.scala]: Ok. let me check

**Pull request merged**